### PR TITLE
Automatic hotel staff antigrief

### PR DESCRIPTION
**Pull request by WhyIsCaeciliusTaken**
## About The Pull Request

Hotel staff now start with a super exile implant, which messages admins when they try to enter the station Z-level, then dusts them. The implant also dusts them if they attempt to remove it via self-surgery

## Why It's Good For The Game

Makes it impossible for hotel staff to grief the station, cuts down on things admins have to do. which is good.

## Changelog 
:cl: WhyIsCaeciliusTaken
admin: Hotel staff now instantly die if they go to the station
/:cl:

### CODE REVIEW

**Comment by Crossedfall** [comment]
I feel like there should be a warning and a countdown to give the 'escaping' player a chance to head back to the hotel.

**Comment by WhyIsCaeciliusTaken** [comment]
> I feel like there should be a warning and a countdown to give the 'escaping' player a chance to head back to the hotel.

Hotel staff don't have teleporter access and so the countdown would almost certainly reach 0 before they managed to gain access to return to the hotel.

File: code/game/objects/items/implants/implant_exile.dm
Comment (PowerfulBacon) [inline code/game/objects/items/implants/implant_exile.dm]: I do not see a source var defined anywhere

Comment (Crossedfall) [inline code/game/objects/items/implants/implant_exile.dm]:
```suggestion
		message_admins("[ADMIN_LOOKUPFLW(source)] tried to enter the station as hotel staff and was dusted.")
```

`source` isn't defined in within this datum. I think you meant to have `AM` here?

Comment (Crossedfall) [reply code/game/objects/items/implants/implant_exile.dm]: daannng just beat me to this

Comment (PowerfulBacon) [reply code/game/objects/items/implants/implant_exile.dm]: I win by 2 seconds 🥇

Comment (WhyIsCaeciliusTaken) [reply code/game/objects/items/implants/implant_exile.dm]: Oops I'll fix that. Yes it's supposed to be AM

**Review by Crossedfall — DISMISSED** [review]
You really should use variable names that are a bit clearer. `source` and `AM` don't seem to make much sense within the context of your additions

File: code/game/objects/items/implants/implant_exile.dm
Comment (WhyIsCaeciliusTaken) [reply code/game/objects/items/implants/implant_exile.dm]: ahem typo. Oops.

Comment (WhyIsCaeciliusTaken) [reply code/game/objects/items/implants/implant_exile.dm]: Well it's now Hotelstaff

File: code/game/objects/items/implants/implant_exile.dm
Comment (MCterra10) [inline code/game/objects/items/implants/implant_exile.dm]: Not technically necessary, but close your span tags. Practice good HTML.

File: code/game/objects/items/implants/implant_exile.dm
Comment (MCterra10) [inline code/game/objects/items/implants/implant_exile.dm]: Here too

**Comment by sheldoniscool** [comment]
Super don't like this.
It's a fun mechanic and you need to be significantly robust to pull of getting onto the station with any of the spawner roles. All the other spawner roles have significant draw backs I.e golems.

I don't even know how to get on the station with hotel staff even though I know must everything in the game. How often are people 'griefing' because it's existed for years and I can count on one hand how many times I've seen hotel staff appear on station.

**Comment by PowerfulBacon** [comment]
> I don't even know how to get on the station with hotel staff

You literally walk into the teleporter you spawn next to

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
See MCTerra's review

**Review by PowerfulBacon — APPROVED** [review]
Seems good, I just want to double check the message admins works after dusting to make sure before merging

**Comment by PowerfulBacon** [comment]
@Crossedfall Needs your approval because of changes requested

**Comment by qwertyquerty** [comment]
@Crossedfall

**Pull request merged**
